packages/svelte/src/compiler/phases/utils.js

```diff
@@ -0,0 +1,33 @@
+/** @import { AST } from '#compiler' */
+
+/**
+ * @param {AST.Attribute["value"]} value
+ */
+export function is_inlinable_expression(value) {
+	if (value === true) return false; // not an expression
+	let nodes = Array.isArray(value) ? value : [value];
+	let has_expression_tag = false;
+	for (let value of nodes) {
+		if (value.type === 'ExpressionTag') {
+			if (!value.metadata.expression.can_inline) {
+				return false;
+			}
+
+			if (
+				// we need to special case null and boolean values because
+				// we want to set them programmatically
+				value.expression.type === 'Literal' &&
+				(value.expression.value === null ||
+					typeof value.expression.value === 'boolean' ||
```

Comment:
this special case fails — change `attribute-boolean-case-insensitive`, for example:

```diff
-<input READONLY={true} REQUIRED={false}>
+<input READONLY={!0} REQUIRED={!1}>
```

Needless to say the same thing happens in this case:

```svelte
<script context="module">
	let t = true;
	let f = false;
</script>

<input READONLY={t} REQUIRED={f}>
```

Bottom line, you can't do this by statically analysing the value. I suspect we could do it based on the name of the attribute, but that logic would live outside this function